ast-nodes/routine-declaration.dart

```diff
@@ -145,6 +145,15 @@ class RoutineDeclaration extends Declaration implements ScopeCreator {
       )
     );
 
+    for (var i = 0; i < this.parameters.length; i++) {
+      var parameter = llvm.LLVMGetParam(routine, i);
+      parameters[i].scopeMark.resolve(parameters[i].name).valueRef = parameter;
```

Comment:
This will not resolve the parameter. Remember how we said that the scope mark of an entity points to the last valid declaration __before__ it? Here you should be resolving from `this.scopes[0].lastChild`